tests/types.test.tsx

```diff
@@ -0,0 +1,32 @@
+import { expectType } from 'ts-expect'
+import { atom, Atom, WritableAtom } from '../src/index'
+import { WithInitialValue } from '../src/core/atom'
+
+it('atom() should return the correct types', () => {
+  function Component() {
+    // primitive atom
+    const primitiveAtom = atom(0)
+    expectType<Atom<number> & WithInitialValue<number>>(primitiveAtom)
```

Comment:
Either,
```suggestion
    expectType<Atom<number> & { init: number }>(primitiveAtom)
```

Or,
```suggestion
    expectType<Atom<number>>(primitiveAtom)
```

What do you think?

Reply:
Also, we want to expect with `PrimitiveAtom` which is public type.